web/src/recent_topics_ui.js

```diff
@@ -541,6 +541,12 @@ export function filters_should_hide_topic(topic_data) {
     }
 
     if (!filters.has("include_muted") && topic_data.type === "stream") {
+        // We want to show the unmuted topics within muted streams in the recent topics.
+        const topic_unmuted = Boolean(user_topics.is_topic_unmuted(msg.stream_id, msg.topic));
+        if (topic_unmuted) {
+            return false;
+        }
```

Comment:
Can you pull this change out to a separate PR? It looks correct to me and would be nice to merge, since the rest of this PR may be bogged down given that I don't see the other comments as being easy to address.

Reply:
Sure working on it.

Reply:
here is the PR: https://github.com/zulip/zulip/pull/25132